web3/_utils/filters.py

```diff
@@ -260,3 +274,29 @@ def match_fn(w3: "Web3", match_values_and_abi: Collection[Tuple[str, Any]], data
             return False
 
     return True
+
+
+@curry
+def select_filter_method(
+    value: Union[str, FilterParams, HexStr],
+    if_new_block_filter: RPCEndpoint,
+    if_new_pending_transaction_filter: RPCEndpoint,
+    if_new_filter: RPCEndpoint,
+) -> RPCEndpoint:
+
+    if is_string(value):
+        if value == "latest":
+            return if_new_block_filter
+        elif value == "pending":
+            return if_new_pending_transaction_filter
+        elif is_hex(value):
+            raise _UseExistingFilter(value)
+        else:
+            raise ValidationError("Filter argument needs to be either 'latest',"
+                                  " 'pending', or a hex-encoded filter_id")
+    elif isinstance(value, dict):
+        return if_new_filter
+    else:
+        raise ValidationError("Filter argument needs to be either the string "
+                              "'pending' or 'latest', a filter_id, "
+                              "or a filter params dictionary")
```

Comment:
When someone encounters either of these `ValidationError` cases, it tends to be useful to include both this information about what was expected, but also to include information about what we actually got.  It can be helpful to include both the `repr(value)` and `type(value)` in the error message.